router/core/router.go

```diff
@@ -871,6 +856,42 @@ func (r *Router) newServer(ctx context.Context, routerConfig *nodev1.RouterConfi
 	return ro, nil
 }
 
+func (r *Router) buildHTTPRouter() http.Handler {
```

Comment:
This HTTP router now has 0 dependencies on the active server or its config. In the next PR I will modify the API to only call this once and move the HTTP server into `cmd.Main`.